### DISCUSSION TITLE
why doesn't my dApp connect mobile metamask app from mobile browser

### DISCUSSION THREAD

**Discussion by sniper365** [rainbow-me/rainbowkit#1294] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

1.0.0

### wagmi Version

1.0.5

### Current Behavior

This is my index.tsx

```
import * as React from "react"
import { ColorModeScript } from "@chakra-ui/react"
import { createRoot } from 'react-dom/client';
import { App } from "./App"
import './index.css';
import merge from 'lodash.merge';
import { getDefaultWallets, RainbowKitProvider, darkTheme, Theme } from '@rainbow-me/rainbowkit';
import { configureChains, createConfig, WagmiConfig } from 'wagmi';
import { goerli, mainnet } from 'wagmi/chains';
import { publicProvider } from 'wagmi/providers/public';
import '@rainbow-me/rainbowkit/styles.css'

const { chains, publicClient, webSocketPublicClient } = configureChains(
  [
    mainnet,
  ],
  [publicProvider()]
);

const { connectors } = getDefaultWallets({
  appName: 'bitoro',
  projectId: '2023-5',
  chains,
});

const wagmiConfig = createConfig({
  autoConnect: true,
  connectors,
  publicClient,
  webSocketPublicClient,
});

const myTheme = merge(darkTheme({
  borderRadius: 'small'
}), {
  colors: {
    accentColor: '#194a8c',  //bitoro.600
    connectButtonBackground: '#00204c',
    modalBackground: '#002559',  //bitoro.1000
    modalText: 'white',
    actionButtonSecondaryBackground: '#6686b2',  //bitoro.300
  },
  fonts: {
    body: '20px',
  },
});

const container = document.getElementById("root")
const root = createRoot(container!)
root.render(
  // <React.StrictMode>
  <>
    <WagmiConfig config={wagmiConfig}>
      <RainbowKitProvider
        chains={chains}
        theme={myTheme}
      >
        <ColorModeScript />
        <App />
      </RainbowKitProvider>
    </WagmiConfig>
  </>
  // </React.StrictMode>,
)
```

It's working on browser but not working on mobile.

Please help me. I searched many many times but not helpful. Thank you

**Comment by DanielSinclair** [comment]
@sniper365 Can you share a CodeSandbox and step by step test path? Please also describe the issue you are seeing very clearly.

**Comment by sniper365** [comment]
@DanielSinclair 

Here is sandbox

https://codesandbox.io/s/github/sniper365/bitoro-codesandbox-test

Please review it?

**Comment by DanielSinclair** [comment]
@sniper365 Am encountering an error with the CodeSandbox:
<img width="516" alt="Screenshot 2023-06-13 at 10 04 05 PM" src="https://github.com/rainbow-me/rainbowkit/assets/4412473/2a6727f0-8b62-48ff-b778-f4ccb9ffbe54">

**Comment by sniper365** [comment]
@DanielSinclair 
Please check it out now. I fixed error.

**Comment by DanielSinclair** [comment]
@sniper365 Seeing build errors. Can you create as minimal of a reproducible example as possible and share the test path?

**Comment by sniper365** [comment]
@DanielSinclair 

I removed some unnecessary modules and did build. 
https://codesandbox.io/s/github/sniper365/bitoro-codesandbox-test
After that it's working. I mean mobile connecting is working

However in vercel project - https://bitoro-codesandbox-test.vercel.app from same git repository, mobile connecting not working.. What's the difference? What did I miss?


And also with same configuration about rainbowkit and wagmi my project running on AWS is also not working.  Really Very weird..

**Comment by DanielSinclair** [comment]
Looks like you've forgotten to include the polyfills, which will create unexpected behavior in newer versions of Create React App that rely on Webpack v5. Reference `polyfills.ts` and the `import './polyfills';` in this [CRA Example](https://codesandbox.io/p/sandbox/github/rainbow-me/rainbowkit/tree/main/examples/with-create-react-app) and the [docs](https://www.rainbowkit.com/docs/installation#react-nextjs-and-webpack) for more information.

**Reply by DanielSinclair** [reply]
This is quiet failure within WalletConnect's libraries, and the pollyfill requirements have been addressed in future versions of RainbowKit that rely on WalletConnect v2.

**Comment by FranRom** [comment]
Hi there, 
I used the polyfills approach, working on 0.12 version but not in the last one
Vite framework. Do you know something else I could try? 
Will WalletConnect works in 0.12 once they shutdown 1.0 version?

**Reply by DanielSinclair** [reply]
[0.12.15](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%400.12.15) and above do now support WalletConnect v2 to make it easier for devs that haven't yet upgraded to Wagmi v1.

**Reply by adrianotadao** [reply]
@DanielSinclair how do I know if rainbowkit is working with wallet connect v2 or not?


is this enough?

```javascript
const { connectors } = getDefaultWallets({
  appName: 'RainbowKit App',
  projectId: 'xxx',
  chains: [],
});
```

UPDATE: I want to make sure that nothing will stop working on Wednesday

**Reply by DanielSinclair** [reply]
@adrianotadao [0.12.15](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%400.12.15) and [1.0.2](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%401.0.2) and above rely on WalletConnect v2 by default.

**Comment by gsw85** [comment]
for mobile, I think metamask change the way it connect, you need to use metamask browser to open your dapps

**Reply by sniper365** [reply]
So connecting metamask app on mobile won't be working any more?

**Reply by DanielSinclair** [reply]
@gsw85 MetaMask is working on a fix for their WalletConnect v2 support, but the dApp browser will work as a fallback for the time being!

**Reply by gsw85** [reply]
I see thanks !

**Reply by DanielSinclair** [reply]
@gsw85 [0.12.17](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%400.12.17) and [1.0.5](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%401.0.5) are now available with a fix for this MetaMask issue.

**Comment by DanielSinclair** [comment]
[0.12.17](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%400.12.17) and [1.0.5](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%401.0.5) are now available with a fix for this MetaMask issue.

**Reply by sniper365** [reply]
```
import * as React from "react"
import { Provider } from 'react-redux';
import { I18nextProvider } from 'react-i18next'
import { ChakraProvider } from '@chakra-ui/react'
import { connectorsForWallets, darkTheme, getDefaultWallets, RainbowKitProvider, Theme } from '@rainbow-me/rainbowkit';
import {
  injectedWallet,
  rainbowWallet,
  walletConnectWallet,
  coinbaseWallet,
  trustWallet,
  metaMaskWallet
} from '@rainbow-me/rainbowkit/wallets';
import { configureChains, createConfig, WagmiConfig } from 'wagmi';
import { mainnet } from 'wagmi/chains';
import { publicProvider } from 'wagmi/providers/public';
import { store } from './state/store';
import { Router } from './Router'
import theme from './theme'
import i18n from './i18n'
import useLanguage from "./hooks/useLanguage";
import '@rainbow-me/rainbowkit/styles.css'
import merge from "lodash.merge";

const ALCHEMY_API_KEY = process.env.REACT_APP_ALCHEMY_API_KEY || ''

const { chains, publicClient, webSocketPublicClient } = configureChains(
  [
    mainnet
  ],
  [publicProvider()]
);

// const { connectors } = getDefaultWallets({
//   appName: 'bitoro',
//   projectId: 'bbbitoro',
//   chains,
// });

const connectors = connectorsForWallets([
  {
    groupName: 'Recommended',
    wallets: [
      injectedWallet({ chains }),
      metaMaskWallet({ projectId: 'bitoro', chains }),
      walletConnectWallet({ projectId: 'bitoro', chains }),
      coinbaseWallet({ appName: 'bitoro', chains }),
      trustWallet({ projectId: 'bitoro', chains }),
    ],
  },
]);

const wagmiConfig = createConfig({
  autoConnect: true,
  connectors,
  publicClient,
  webSocketPublicClient,
});

const myTheme = merge(darkTheme({
  borderRadius: 'small'
}), {
  colors: {
    accentColor: '#194a8c',  //bitoro.600
    connectButtonBackground: '#00204c',
    modalBackground: '#002559',  //bitoro.1000
    modalText: 'white',
    actionButtonSecondaryBackground: '#6686b2',  //bitoro.300
  },
  fonts: {
    body: '20px'
  }
});

export const App = () => {
  const { language } = useLanguage()

  i18n.changeLanguage(language)

  return (
    <I18nextProvider i18n={i18n}>
      <WagmiConfig config={wagmiConfig}>
        <RainbowKitProvider
          chains={chains}
          theme={myTheme}
        >
          <ChakraProvider theme={theme}>
            <Provider store={store}>
              <Router />
            </Provider>
          </ChakraProvider>
        </RainbowKitProvider>
      </WagmiConfig>
    </I18nextProvider>
  )
}
```

@DanielSinclair 
I tried to config with Metamask, WalletConnect, Coinbase, TrustWallet after install updated [1.0.5](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%401.0.5) but Not succeed. There are no reaction when I click wallets on mobile. what do you think about this?

**Reply by DanielSinclair** [reply]
@sniper365 Your `projectId` is invalid. You need to obtain a valid key from [WalletConnect Cloud](https://cloud.walletconnect.com/), otherwise requests to WalletConnect will fail.

**Reply by sniper365** [reply]
`projectId` is only for walletConnectWallet or others like metamaskWallet or trustWallet?

**Reply by DanielSinclair** [reply]
@sniper365 It is required by all WalletConnect connectors (that redirect to a wallet on mobile, or support a qr code on desktop)

**Reply by sniper365** [reply]
One more, @DanielSinclair (not sure do I have to ask even this but hope your help)
![image](https://github.com/rainbow-me/rainbowkit/assets/30114178/a7ef698a-b6c1-4041-af30-15f7ef4a7cf5)
![image](https://github.com/rainbow-me/rainbowkit/assets/30114178/f8569aec-e40c-4a04-b9d3-f57f96c40bdb)

As you said, even metamaskWallet need `projectId`, I got the `projectId` after signin and added to my chains.
```
const connectors = connectorsForWallets([
  {
    groupName: 'Recommended',
    wallets: [
      injectedWallet({ chains }),
      metaMaskWallet({ projectId: 'a7e8f5ac54...80e865870', chains }),
      walletConnectWallet({ projectId: 'a7e8f5ac54...80e865870', chains }),
      coinbaseWallet({ appName: 'bitoro', chains }),
      trustWallet({ projectId: 'a7e8f5ac54...80e865870', chains }),
    ],
  },
]);
```
then Do I have to fill [walletConnect Cloud]
(https://cloud.walletconnect.com/) form? After submitting and pending review, I got above messages

**Reply by DanielSinclair** [reply]
Yes, you need to create a Project on WalletConnect Cloud. Please relay any questions regarding this to WalletConnect.

**Reply by sniper365** [reply]
Ok thanks

**Reply by sniper365** [reply]
![image](https://github.com/rainbow-me/rainbowkit/assets/30114178/927978a3-0edf-4687-ba63-89b69d89eb8c)
@DanielSinclair At this moment, trustWallet and coinbaseWallet working but metamask and walletConnect not working... Wondering if you have ideas

**Reply by adrianotadao** [reply]
>  but metamask and walletConnect not working... Wondering if you have ideas

@sniper365 what is happening to metamask? Have you added your domain to **Allowed Domains** section yet?

**Reply by gsw85** [reply]
your walletconnect should be working, the pending review here is for if you can be listed in Walletconnect's explorer, if you failed review, simply means you won't be listed in Walletconnect explorer

**Comment by gsw85** [comment]
on this part:

```
// const { connectors } = getDefaultWallets({
//   appName: 'bitoro',
//   projectId: 'bbbitoro',
//   chains,
// });

const connectors = connectorsForWallets([
  {
    groupName: 'Recommended',
    wallets: [
      injectedWallet({ chains }),
      metaMaskWallet({ projectId: 'bitoro', chains }),
      walletConnectWallet({ projectId: 'bitoro', chains }),
      coinbaseWallet({ appName: 'bitoro', chains }),
      trustWallet({ projectId: 'bitoro', chains }),
    ],
  },
]);

```

try

```
const projectId = process.env.WALLETCONNECT_PROJECT_ID;  // replace with your walletconnect project id


const { wallets } = getDefaultWallets({
  appName: "Your app name",
  projectId,
  chains,
});


const connectors = connectorsForWallets([
  ...wallets, 
  {
    groupName: 'Recommended',
    wallets: [
      trustWallet({ projectId, chains }),
    ],
  },
]);
```

in your code, your projectId is not group name , you will need to get from walletconnect.

**Reply by sniper365** [reply]
Hey @gsw85 thanks for your kind answer...
#index.tsx
```
import './polyfills';
import * as React from "react"
import { createRoot } from 'react-dom/client';
import reportWebVitals from './reportWebVitals';
import merge from "lodash.merge";
import { I18nextProvider } from 'react-i18next'
import { connectorsForWallets, darkTheme, getDefaultWallets, RainbowKitProvider, Theme } from '@rainbow-me/rainbowkit';
import { trustWallet } from '@rainbow-me/rainbowkit/wallets';
import { configureChains, createConfig, WagmiConfig } from 'wagmi';
import { mainnet } from 'wagmi/chains';
import { publicProvider } from 'wagmi/providers/public';
import i18n from './i18n'
import App from './App'
import './index.css';
import '@rainbow-me/rainbowkit/styles.css'

const projectId = process.env.REACT_APP_WALLETCONNECT_PROJECT_ID || ''

const { chains, publicClient, webSocketPublicClient } = configureChains(
  [
    mainnet
  ],
  [publicProvider()]
);

const { wallets } = getDefaultWallets({
  appName: "bitoro",
  projectId,
  chains,
});

const connectors = connectorsForWallets([
  ...wallets,
  {
    groupName: 'Recommended',
    wallets: [
      trustWallet({ projectId, chains }),
    ],
  },
]);

const wagmiConfig = createConfig({
  autoConnect: true,
  connectors,
  publicClient,
  webSocketPublicClient,
});

const myTheme = merge(darkTheme({
  borderRadius: 'small'
}), {
  colors: {
    accentColor: '#194a8c',  //bitoro.600
    connectButtonBackground: '#00204c',
    modalBackground: '#002559',  //bitoro.1000
    modalText: 'white',
    actionButtonSecondaryBackground: '#6686b2',  //bitoro.300
  },
  fonts: {
    body: '20px'
  }
});

const container = document.getElementById("root")
const root = createRoot(container!)
root.render(
  // <React.StrictMode>
  <I18nextProvider i18n={i18n}>
    <WagmiConfig config={wagmiConfig}>
      <RainbowKitProvider
        chains={chains}
        theme={myTheme}
      >
        <App />
      </RainbowKitProvider>
    </WagmiConfig>
  </I18nextProvider>
  // </React.StrictMode>,
)

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
```

[video1.zip](https://github.com/rainbow-me/rainbowkit/files/12039018/video1.zip)

No action from metamask.. what's wrong?

**Reply by gsw85** [reply]
probably try to do it at app.tsx instead,, just a guess for trial and error

**Reply by sniper365** [reply]
I tried twice but not succeed. I reinstalled node_modules with rainbow@1.0.5 and wagmi@1.3.8
